### PR TITLE
test(numeral-date): convert cypress tests to playwright

### PR DESCRIPTION
**Pull request by nuria1110**
### Proposed behaviour

Refactor tests to use Playwright

### Current behaviour

Tests currently use Cypress

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The following CodeSandbox is an example of the broken behaviour.
You can see the new behaviour by looking at the version in the comment by `codesandbox[bot]`.

### CODE REVIEW

File: playwright/components/numeralDate/index.ts
Comment (edleeks87) [inline playwright/components/numeralDate/index.ts:1]: comment: lets keep the same naming convention here `numeralDate` -> `numeral-date`

File: playwright/components/numeralDate/locators.ts
Comment (edleeks87) [inline playwright/components/numeralDate/locators.ts:1]: comment: `numeralDate` -> `numeral-date`

File: playwright/components/numeralDate/index.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import type { Page } from "@playwright/test";
```
Comment (edleeks87) [inline playwright/components/numeralDate/index.ts:1]:
```suggestion
import { Page } from "@playwright/test";
```

File: src/components/numeral-date/numeral-date.pw.tsx [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+  NumeralDateComponent,
+  NumeralDateControlled,
+} from "./components.test-pw";
+
+import {
+  numeralDateComponent,
+  numeralDateInput,
+} from "../../../playwright/components/numeralDate";
+
+import Box from "../../../src/components/box";
```
Comment (edleeks87) [inline src/components/numeral-date/numeral-date.pw.tsx:15]:
```suggestion
import Box from "../box";
```

File: src/components/numeral-date/numeral-date.pw.tsx [hunk 3]
```diff
@@ -0,0 +1,685 @@
...
+      mount,
+      page,
+    }) => {
+      await mount(<NumeralDateComponent dateFormat={dateFormat} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with enableInternalError prop", async ({
```
Comment (edleeks87) [inline src/components/numeral-date/numeral-date.pw.tsx:605]: comment: You probably want to trigger the validation tooltip here and then run the check

File: src/components/numeral-date/numeral-date.pw.tsx [hunk 4]
```diff
@@ -0,0 +1,685 @@
...
+    mount,
+    page,
+  }) => {
+    await mount(<NumeralDateControlled enableInternalError />);
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with enableInternalWarning prop", async ({
```
Comment (edleeks87) [inline src/components/numeral-date/numeral-date.pw.tsx:614]: comment: You probably want to trigger the validation tooltip here and then run the check

File: src/components/numeral-date/numeral-date.pw.tsx [hunk 5]
```diff
@@ -0,0 +1,685 @@
...
+        labelAlign="right"
+        labelWidth={30}
+        adaptiveLabelBreakpoint={960}
+      />
+    );
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with labelHelp prop", async ({
```
Comment (edleeks87) [inline src/components/numeral-date/numeral-date.pw.tsx:650]: comment: You probably want to trigger the help tooltip here and then run the check

File: src/components/numeral-date/numeral-date.pw.tsx [hunk 6]
```diff
@@ -0,0 +1,685 @@
...
+    page,
+  }) => {
+    await mount(
+      <NumeralDateControlled labelInline labelAlign="right" labelWidth={30} />
+    );
+
+    await checkAccessibility(page);
+  });
+
+  test("should pass accessibility tests for NumeralDateControlled component with adaptive behaviour", async ({
```
Comment (edleeks87) [inline src/components/numeral-date/numeral-date.pw.tsx:634]: question: have we confirmed the viewport size is set to trigger the adaptive behaviour here?

**Review by edleeks87 — COMMENTED** [review]
@nuria1110 well done for your first playwright refactor. You'll need to also delete the [cypress locator files](https://github.com/Sage/carbon/blob/01244b105f85fbf7001710ce71d9abb0d3ef46e5/cypress/locators/numeralDate/index.js). I've also left some minor comments/ questions give me a shout if anything is unclear

File: src/components/numeral-date/numeral-date.pw.tsx [hunk 7]
```diff
@@ -0,0 +1,711 @@
+import React from "react";
+import { test, expect } from "@playwright/experimental-ct-react17";
+
+import NumeralDate, { NumeralDateProps } from ".";
+import {
+  NumeralDateComponent,
+  NumeralDateControlled,
+} from "./components.test-pw";
```
Comment (stephenogorman) [inline src/components/numeral-date/numeral-date.pw.tsx:8]: You can remove NumeralDateComponent from "./numeral-date-test.stories.tsx" You can also remove line 17 from the 'export default' section of the same file:
`  excludeStories: ["NumeralDateComponent"],`

File: src/components/numeral-date/numeral-date.pw.tsx [hunk 8]
```diff
@@ -0,0 +1,711 @@
...
+
+import { ICON } from "../../../playwright/components/locators";
+import {
+  CHARACTERS,
+  SIZE,
+  VALIDATION,
+} from "../../../playwright/support/constants";
+
+const testData = [CHARACTERS.DIACRITICS, CHARACTERS.SPECIALCHARACTERS];
+
```
Comment (stephenogorman) [inline src/components/numeral-date/numeral-date.pw.tsx:39]: Should these test be within a `test.describe`? The cypress tests were broken up into prop, event and accessibility tests.

**Review by edleeks87 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by stephenogorman — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Pull request merged**